mica/starcheck/starcheck.py

```diff
@@ -26,52 +28,167 @@
+                         server='/proj/sot/ska/data/cmd_states/cmd_states.db3')
+    return timelines_db.fetchone(
         "select * from timeline_loads where datestop >= '%s' "
         " and datestart <= '%s' and scs <= 130 order by datestart desc"
         % (date, date))
 
-def get_mp_dir(obsid, config=None):
+
+def get_flickpix_mons(start=None, config=None):
     if config is None:
         config = DEFAULT_CONFIG
-    aca_db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read')
-    possible_runs = aca_db.fetchall(
-        "select * from tl_obsids where obsid = %d" % obsid)
-    if not len(possible_runs):
-        db = Ska.DBI.DBI(dbi='sqlite', server=config['server'])
-        sc = db.fetchall(
+    db = Ska.DBI.DBI(dbi='sqlite', server=config['server'])
+    # I'm ignoring the weird case of SIR obsids or whatever where the following ER keeps
+    # the MON
+    mons = db.fetchall("""select obsid, mp_starcat_time, type, sz, yang, zang, dir
+                          from starcheck_catalog, starcheck_id
+                          where type = 'MON' and obsid > 40000
+                          and starcheck_id.id = starcheck_catalog.sc_id""")
+    mons = Table(mons)
+    if start is not None:
+        mons = mons[mons['mp_starcat_time'] >= DateTime(start).date]
+    statuses = []
+    # Check which ones actually ran or are likely to run
+    for mon in mons:
+        print mon
+        try:
+            actual_catalog = get_catalog_at_date(mon['mp_starcat_time'])
+            if (actual_catalog is not None and actual_catalog['obs']['obsid'] == mon['obsid']
+                    and actual_catalog['mp_dir'] == mon['dir']):
+                statuses.append(actual_catalog['status'])
+            else:
+                statuses.append('none')
+        except LookupError:
+            statuses.append('none')
+    # Add that status info to the table and filter on it
+    mons['status'] = statuses
+    mons = mons[mons['status'] != 'none']
+    return mons
+
+
+def get_catalog_at_date(date, config=None, timelines_db=None):
+    """
+    For a given date, return a dictionary describing the starcheck catalog that should apply.
+    The content of that dictionary is from the database tables that parsed the starcheck report.
+    A catalog is defined as applying, in this function, to any time from the end of the
+    previous dwell through the end of the dwell in which the catalog was used.
+    """
+
+    if config is None:
+        config = DEFAULT_CONFIG
+    date = DateTime(date).date
+    if timelines_db is None:
+        timelines_db = Ska.DBI.DBI(dbi='sqlite',
+                                   server='/proj/sot/ska/data/cmd_states/cmd_states.db3')
+    last_tl = timelines_db.fetchone(
+        "select max(datestop) as datestop from timelines")['datestop']
+    first_tl = timelines_db.fetchone(
+        "select min(datestart) as datestart from timelines")['datestart']
+    db = Ska.DBI.DBI(dbi='sqlite', server=config['server'])
```

Comment:
All through these changes, I'm not sure how much I really need to use the database context manager.